internal/utils/sqlite/sqlite.go

```diff
@@ -0,0 +1,116 @@
+package sqlite
+
+import (
+	"database/sql"
+	"database/sql/driver"
+	"sync"
+
+	"github.com/mattn/go-sqlite3"
+)
+
+// Init creates the connections map and registers the driver with the SQL package.
+func init() {
+	conns = make(map[uint64]*Conn)
+
+	sql.Register(DriverName, &Driver{})
+}
+
+// In order to use this driver, specify the DriverName to sql.Open.
+const (
+	DriverName = "ensign_sqlite3"
+)
+
+var (
+	seq   uint64
+	mu    sync.Mutex
+	conns map[uint64]*Conn
+)
+
+// Driver embeds a sqlite3 driver but overrides the Open function to ensure the
+// connection created is a local connection with a sequence ID. It then maintains the
+// connection locally until it is closed so that the underlying sqlite3 connection can
+// be returned on demand.
+type Driver struct {
+	sqlite3.SQLiteDriver
+}
+
+// Open implements the sql.Driver interface and returns a sqlite3 connection that can
+// be fetched by the user using GetLastConn. The connection ensures it's cleaned up
+// when it's closed. This method is not used by the user, but rather by sql.Open.
+func (d *Driver) Open(dsn string) (_ driver.Conn, err error) {
+	var inner driver.Conn
+
+	if inner, err = d.SQLiteDriver.Open(dsn); err != nil {
+		return nil, err
+	}
+
```

Comment:
Let's be consistent with our function signatures, theres no need to just name the error return

```suggestion
func (d *Driver) Open(dsn string) (driver.Conn, error) {
	inner, err := d.SQLiteDriver.Open(dsn); 
	if err != nil {
		return nil, err
	}
```